### PR TITLE
ui for private projects

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

**Comment by stas-vilchik** [comment]
@teryk please review only 126a560

File: server/sonar-web/src/main/js/apps/projects-admin/search.js [hunk 1]
```diff
@@ -144,7 +155,10 @@ export default class Search extends React.PureComponent {
                   />
                 </form>
               </td>
-              <td className="thin text-middle">
+              <td className="thin nowrap text-middle">
+                <button className="spacer-right" onClick={this.bulkApplyTemplate}>
+                  Bulk Apply Permission Template
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/projects-admin/search.js]: Missing translation

File: server/sonar-web/src/main/js/apps/projects-admin/ChangeVisibilityForm.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                </a>
+              </p>
+              <p className="text-muted spacer-top" style={{ paddingLeft: 22 }}>
+                {translate('visibility', visibility, 'description.short')}
+              </p>
+            </div>
+          ))}
+
+          <div className="alert alert-warning ">
+            {translate('organization.change_visibility_form.warning')}
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/projects-admin/ChangeVisibilityForm.js]: I don't know if it's on purpose or not but there was a warning icon on the mock up

File: server/sonar-web/src/main/js/apps/permissions/project/components/App.js [hunk 3]
```diff
@@ -17,39 +17,357 @@
...
+  handleFilterChange = (filter: string) => {
+    if (this.mounted) {
+      this.setState({ filter }, this.loadHolders);
+    }
+  };
+
+  handleQueryChange = (query: string) => {
+    if (this.mounted) {
+      this.setState({ query }, () => {
+        if (query.length === 0 || query.length > 2) {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/permissions/project/components/App.js]: I'd keep it for readability

File: server/sonar-web/src/main/js/apps/projects-admin/ChangeVisibilityForm.js (hunk 2 above)
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects-admin/ChangeVisibilityForm.js]: I've reused already existing component. I don't want to change rendering of every warning in this PR.

**Comment by stas-vilchik** [comment]
@gregaubert I've applied your feedback and pushed the last commit.

**Comment by stas-vilchik** [comment]
> There is something wrong when editing permissions templates, I'm not able to search or filter on users or groups

The problem existed even before this PR, I will fix it separately.

**Pull request merged**